internal/wait/noderesourcetopologies.go

```diff
@@ -110,3 +113,65 @@ func (wt Waiter) ForNodeResourceTopologiesSettled(ctx context.Context, threshold
 	klog.Infof("done waiting for NRT to stabilize; observed nodes=%d ready=%d", pfpState.Len(), pfpState.CountReady(threshold))
 	return &nrtList, err
 }
+
+type NRTFilterFunc func(nrt *nrtv1alpha2.NodeResourceTopology) bool
+
+func NRTPassthrough(nrt *nrtv1alpha2.NodeResourceTopology) bool {
+	return true
+}
+
+func (wt Waiter) ForNodeResourceTopologiesEqualTo(ctx context.Context, nrtListReference *nrtv1alpha2.NodeResourceTopologyList, nrtFilter NRTFilterFunc) (nrtv1alpha2.NodeResourceTopologyList, error) {
+	var updatedNrtList nrtv1alpha2.NodeResourceTopologyList
+	klog.Infof("Waiting up to %v to have %d NRT objects restored", wt.PollTimeout, len(nrtListReference.Items))
+	err := k8swait.Poll(wt.PollInterval, wt.PollTimeout, func() (bool, error) {
+		err := wt.Cli.List(ctx, &updatedNrtList)
+		if err != nil {
+			klog.Errorf("cannot get the NRT List: %v", err)
+			return false, err
+		}
+		for idx := range nrtListReference.Items {
+			referenceNrt := &nrtListReference.Items[idx]
+
+			if !nrtFilter(referenceNrt) {
```

Comment:
nit: Readability-wise I would expect to read it as 
```
if nrtFilter(referenceNrt) {
    klog.Warningf("skipping NRT %q because of filter", referenceNrt.Name)
    continue
}
without the `!` mark 
```

Reply:
I'm not sure I'm getting you right because if I do that's a bug, not a nit :)

Reply:
that's actually a good point. Let me see.